File: src/dotnet/Common/Models/ResourceProviders/Agent/AgentWorkflows/AzureAIAgentServiceAgentWorkflow.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.ResourceProviders.Agent.AgentWorkflows
{
    /// <summary>
    /// Provides an agent workflow configuration for an Azure AI Agent Service workflow.
    /// </summary>
    public class AzureAIAgentServiceAgentWorkflow : AgentWorkflowBase
    {
        /// <inheritdoc/>
        [JsonIgnore]
        public override string Type => AgentWorkflowTypes.AzureAIAgentService;

        /// <summary>
        /// The Azure AI Agent ID for the agent workflow.
        /// </summary>
        [JsonPropertyName("agent_id")]
        public string? AgentId { get; set; }

        /// <summary>
        /// The vector store ID for the agent.
        /// </summary>
        [JsonPropertyName("vector_store_id")]
        public string? VectorStoreId { get; set; }

        /// <summary>
        /// The Azure AI project connection string for the agent workflow.
        /// </summary>
        [JsonPropertyName("project_connection_string")]
        public required string ProjectConnectionString { get; set; }
    }
}

File: src/python/PythonSDK/foundationallm/models/agents/agent_workflows/azure_ai_agent_service_agent_workflow.py
from pydantic import Field
from typing import Any, Optional, Self, Literal
from foundationallm.langchain.exceptions import LangChainException
from foundationallm.utils import object_utils
from .agent_workflow_base import AgentWorkflowBase

class AzureAIAgentServiceAgentWorkflow(AgentWorkflowBase):
    """
    The configuration for an Azure AI Agent Service agent workflow.
    """
    type: Literal["azure-ai-agent-service-workflow"] = "azure-ai-agent-service-workflow"
    agent_id: str = Field(description="The ID of the agent in the Azure AI Agent Service.")
    vector_store_id: Optional[str] = Field(description="The ID of the vector store for the agent.")
    project_connection_string: Optional[str] = Field(description="The connection string for the Azure AI project.")
   
    @staticmethod
    def from_object(obj: Any) -> Self:
        workflow: AzureAIAgentServiceAgentWorkflow = None
        try:
            workflow = AzureAIAgentServiceAgentWorkflow(**object_utils.translate_keys(obj))
        except Exception as e:
            raise LangChainException(f"The Azure AI Agent Service agent workflow object provided is invalid. {str(e)}", 400)
        
        if workflow is None:
            raise LangChainException("The Azure AI Agent Service agent workflow object provided is invalid.", 400)

        return workflow
